pkg/storage/errors.go

```diff
@@ -96,15 +114,24 @@ func IsUnreachable(err error) bool {
 
 // IsTestFailed returns true if and only if err is a write conflict.
 func IsTestFailed(err error) bool {
-	return isErrCode(err, ErrCodeResourceVersionConflicts)
+	return isErrCode(err, ErrCodeResourceVersionConflicts, ErrCodeInvalidUID)
```

Comment:
@hongchaodeng How do you set UID as a precondition in etcd v3? Could you point me to the related code? Thanks.